### PR TITLE
Added --maxMatchingReads to title filtering

### PR DESCRIPTION
**Pull request by terrycojones**
Also: got rid of all calls to `keys` on dictionaries.

### CODE REVIEW

File: test/blast/test_titles.py [hunk 1]
```diff
@@ -552,7 +552,36 @@ def testMinMatchingReads(self):
                 [
                     'gi|887699|gb|DQ37780 Cowpox virus 15',
                 ],
-                list(result.keys()))
+                list(result))
+
+    def testMaxMatchingReads(self):
+        """
+        The filter function work correctly when passed a value for
```
Comment (bamueh) [inline test/blast/test_titles.py:559]: The filter function *must* work correctly...

File: test/diamond/test_titles.py [hunk 2]
```diff
@@ -379,7 +379,36 @@ def testMinMatchingReads(self):
                 [
                     'gi|887699|gb|DQ37780 Cowpox virus 15',
                 ],
-                list(result.keys()))
+                list(result))
+
+    def testMaxMatchingReads(self):
+        """
+        The filter function work correctly when passed a value for
```
Comment (bamueh) [inline test/diamond/test_titles.py:386]: The filter function *must* work correctly

File: test/blast/test_titles.py (hunk 1 above)
Comment (terrycojones) [reply test/blast/test_titles.py]: Fixed (and in many other places) - thanks.

File: test/diamond/test_titles.py (hunk 2 above)
Comment (terrycojones) [reply test/diamond/test_titles.py]: Fixed.

**Pull request merged**